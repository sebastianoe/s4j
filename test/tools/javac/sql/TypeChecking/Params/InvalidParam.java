/*
 * @test
 * @summary Param is not accepted due to its type. We don't know how to put it to the DB.
 * @compile/fail/ref=InvalidParam.out InvalidParam.java -sqlrepo /Users/sebastian/ma/jdk7u-langtools/test/tools/javac/sql/repo.json
 */
// key: compiler.err.sql.validation.param.type.no.jdbctype
import java.sql.ResultSet;
import java.sql.SQLException;

public class InvalidParam {
	public static void main(String[] args) throws SQLException {
		// we can't pass a StringBuilder as an SQL param 
		StringBuilder sb = new StringBuilder();
		ResultSet rs = SQL[SELECT * FROM line_item WHERE quantity > $sb$];
	}
}