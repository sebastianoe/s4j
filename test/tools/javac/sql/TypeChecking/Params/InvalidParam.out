error: The parameter sb does not have JDBC-compatible type: java.lang.StringBuilder.
  Do not know how to pass it to the SQL query.
1 error